src/kakarot/instructions/stop_and_math_operations.cairo

```diff
@@ -250,14 +251,25 @@ namespace StopAndMathOperations {
         jmp end;
 
         SIGNEXTEND:
+        // Examples:
+        // SignExtend 85
+        // x = 01010101, b = 1
+        // Extend x to be two bytes:
+        // x = 00000000 01010101
+        // ---
+        // SignExtend -28
+        // x = 11100100, b = 1
+        // Extend x to be two bytes:
+        // x = 11111111 10101010
+        let bitwise_ptr = cast([fp - 4], BitwiseBuiltin*);
```

Comment:
you can put this in the uint256_signextend utils possibly, surely not here. But I challenge the value. The tests are here for this

Reply:
It feels to me that SignExtend was deeply misunderstood for a long time. 
Adding comments can help solve this problem for more opaque opcodes

Reply:
Modified